apps/browser/src/platform/popup/layout/popup-compact-mode.service.ts

```
@@ -0,0 +1,31 @@
+import { inject, Injectable } from "@angular/core";
+import { map, Observable } from "rxjs";
+
+import { GlobalStateProvider, KeyDefinition, THEMING_DISK } from "@bitwarden/common/platform/state";
+import { CompactModeService } from "@bitwarden/components";
+
+const COMPACT_MODE = new KeyDefinition<boolean>(THEMING_DISK, "compactMode", {
+  deserializer: (s) => s,
+});
+
+/**
+ * Service to persist Compact Mode to state / user settings.
+ **/
+@Injectable({ providedIn: "root" })
+export class PopupCompactModeService implements CompactModeService {
+  private state = inject(GlobalStateProvider).get(COMPACT_MODE);
+
+  enabled$: Observable<boolean> = this.state.state$.pipe(map((state) => state ?? false));
+
+  init() {
+    this.enabled$.subscribe((enabled) => {
+      enabled
+        ? document.body.classList.add("tw-bit-compact")
+        : document.body.classList.remove("tw-bit-compact");
```

do we have to worry about the remove function failing if `tw-bit-compact` doesnt exist? i forget if that fails silently or not

---

It doesn't error: 

> If the string is not in the list, no error is thrown, and nothing happens. 
https://developer.mozilla.org/en-US/docs/Web/API/DOMTokenList/remove